pkg/common-controller/snapshot_controller_base.go

```diff
@@ -243,7 +246,10 @@ func (ctrl *csiSnapshotCommonController) snapshotWorker() {
 			return false
 		}
 		newSnapshot, err := ctrl.checkAndUpdateSnapshotClass(snapshot)
-		if err == nil {
+		if err == nil || errors.IsNotFound(err) {
+			// We should still handle deletion events even if the VolumeSnapshotClass
+			// is not found in the cluster
+			klog.V(5).Infof("deleting snapshot %q ; snapshot has either been scheduled for deletion, or snapshotClass has already been removed", key)
```

Comment:
I think "snapshot has either been scheduled for deletion" can be removed as both condition will go thru "deleteSnapshot".
Suggest change to the following:
deleting snapshot %q; snapshotClass may have already been removed

Reply:
This has been updated to use the suggested message.